R/nneo_wrangle.R

```diff
@@ -0,0 +1,108 @@
+    product_code<-site_code_info$dataProducts$dataProductCode[grep(tolower(gsub("\\(", "",data_var)),
+                              gsub("\\(", "",tolower(site_code_info$dataProducts$dataProductTitle)))]
+    #if empty:
+    if(length(product_code)==0){stop(paste0("data product(s) not available for: ", site_code))}
+    #create year_month variable for nneo_data and nneo_file:
+    year_month<-c(substr(time_start,0,7),substr(time_end,0,7))
+    #check if it's same month - won't need to gather multiple monthly files:
+    if(length(unique(year_month))==1){year_month<-unique(year_month)}
+    #use nneo_data to get available file(s) via user input:
+    var_data<-unlist(lapply(year_month, function(y) lapply(product_code,
+                     function(x) nneo::nneo_data(product_code = x,
+                                           site_code = site_code,
+                                           year_month = y,
+                                           package=package))),recursive = FALSE)
+    #combine data filenames into one df:
+    files_package<-do.call("rbind",lapply(lapply(var_data, "[[", "data"),
+                                            "[[", "files"))
+    #get filenames that match user requested time_agr and sort:
+    searh_terms<-paste0(time_agr,"min.csv|",time_agr,"_minutes.csv")
+    files_time_agr<-sort(files_package$name[grep(searh_terms,files_package$name)])
+    #get the data using nneo_file:
+    data_all<-lapply(year_month, function(y) lapply(unique(files_time_agr),
+                                   function(x) nneo::nneo_file(product_code = substr(x,15,27),
+                                                               site_code = site_code,
+                                                               year_month = y,
+                                                               filename = x)))
+    #name first level of list:
+    names(data_all)<-year_month
+    #rbind dataframes of similar data:
+    data_length<-unique(unlist(lapply(data_all, function(x) length(x))))
+    interim<-list()
+    for(i in 1:data_length){
+      interim[[i]]<-do.call("rbind",lapply(data_all, "[[", i))
+      names(interim[[i]])<-c(names(interim[[i]])[1:2],
+                          paste0(names(interim[[i]][3:length(interim[[i]])]),
+                                 substr(unique(files_time_agr),34,41)[i]))
+    }
+    #merge NEON data then convert startDateTime and endDateTime to POSIX format:
+    data_merge<-Reduce(function(x, y) merge(x, y, by=c("startDateTime","endDateTime")),interim)
+    #remove columns with all NAs:
+    data_clean<-data_merge[, !apply(is.na(data_merge), 2, all)]
+    #final filter by date:
+    date_filt_start<-min(grep(time_start,as.Date(substr(data_clean$startDateTime,0,10))))
+    date_filt_end<-max(grep(time_end,as.Date(substr(data_clean$startDateTime,0,10))))
+    data_final<-data_clean[date_filt_start:date_filt_end,]
+    #output:
+    return(data.frame(data_final))
```

Comment:
@sckott OK, I think it should be an easy fix to convert over to Tibble.  I'll let you know if for some reason I can't do this